test/AsmResolver.Workspaces.DotNet.Tests/AssemblyTraversalTest.cs

```diff
@@ -0,0 +1,268 @@
+        }
+
+        [Fact]
+        public void AssemblyCheck()
+        {
+            var assemblyDefinitions = GetAllMembers<AssemblyDefinition>(_fixture.AllAssemblies, TableIndex.Assembly);
+            TraverseObjects(assemblyDefinitions);
+        }
+
+        [Fact]
+        public void FileCheck()
+        {
+            /* Due to not having any assembly with file references we will use custom generated assembly
+             var fileReferences = GetAllMembers<FileReference>(_fixture.AllAssemblies, TableIndex.File);
+            TraverseObjects(fileReferences);*/
+            var assembly = new AssemblyDefinition("Assembly", new Version(1, 0, 0, 0));
+            var module = new ModuleDefinition("Module");
+
+            var file1 = new FileReference("SomeName1", FileAttributes.ContainsMetadata);
+            var file2 = new FileReference("SomeName2", FileAttributes.ContainsMetadata);
+
+            assembly.Modules.Add(module);
+            module.FileReferences.Add(file1);
+            module.FileReferences.Add(file2);
+
+            var workspace = new DotNetWorkspace();
+            var context = new AnalysisContext(workspace);
+
+            workspace.Assemblies.Add(assembly);
+            context.SchedulaForAnalysis(assembly);
+
+            workspace.Analyze(context);
+
+            Assert.True(context.TraversedObjects.Contains(file1));
+            Assert.True(context.TraversedObjects.Contains(file2));
+        }
+
+        [Fact]
+        public void ExportedTypeCheck()
+        {
+            var exportedTypes = GetAllMembers<ExportedType>(_fixture.AllAssemblies, TableIndex.ExportedType);
+            TraverseObjects(exportedTypes);
+        }
+
+        [Fact]
+        public void ManifestResourceCheck()
+        {
+            var manifestResources = GetAllMembers<ManifestResource>(_fixture.AllAssemblies, TableIndex.ManifestResource);
+            TraverseObjects(manifestResources);
+        }
+
+        [Fact]
+        public void GenericParamCheck()
+        {
+            var genericParameters = GetAllMembers<GenericParameter>(_fixture.AllAssemblies, TableIndex.GenericParam);
+            TraverseObjects(genericParameters);
+        }
+
+        [Fact]
+        public void MethodSpecCheck()
+        {
+            var methodSpecifications = GetAllMembers<MethodSpecification>(_fixture.AllAssemblies, TableIndex.MethodSpec);
+            TraverseObjects(methodSpecifications);
+        }
+
+        [Fact]
+        public void GenericParamConstraintCheck()
+        {
+            var genericParameterConstraints =
+                GetAllMembers<GenericParameterConstraint>(_fixture.AllAssemblies, TableIndex.GenericParamConstraint);
+            TraverseObjects(genericParameterConstraints);
+        }
+
+        private void TraverseObjects(IEnumerable<object> members)
+        {
+            Assert.NotEmpty(members);
+
+            var workspace = new DotNetWorkspace();
+            var context = new AnalysisContext(workspace);
```

Comment:
This should never be instantiated outside from the workspace itself, as described in one of the review points before as it breaks SOLID.

Perhaps it is an idea to make the `Analyze` function return an `AnalysisResult` of some sorts, which does include the traversed objects.